StRoot/StFstHitMaker/StFstHitMaker.cxx

```diff
@@ -0,0 +1,214 @@
+			int  meanRStrip = -1, meanPhiStrip = -1;
+                        float  charge = 0., chargeErr = 0.;
+			unsigned char  maxTb = -1;
+			int	 key = -1;
+			float phiInner = -999.9, phiOuter = -999.9;
+
+			for (std::vector< StFstCluster * >::iterator clusterIter = clusterCollectionPtr->getClusterVec().begin(); clusterIter != clusterCollectionPtr->getClusterVec().end(); ++clusterIter) {
+				idTruth         = (*clusterIter)->getIdTruth();
+				key             = (*clusterIter)->getKey();
+				disk            = (*clusterIter)->getDisk();
+				wedge           = (*clusterIter)->getWedge();
+				sensor          = (*clusterIter)->getSensor();
+				apv             = (*clusterIter)->getApv();
+				meanRStrip      = (*clusterIter)->getMeanRStrip();
+				meanPhiStrip    = (*clusterIter)->getMeanPhiStrip();
+				maxTb           = (*clusterIter)->getMaxTimeBin();
+				charge          = (*clusterIter)->getTotCharge();
+				chargeErr       = (*clusterIter)->getTotChargeErr();
+				nRawHits        = (*clusterIter)->getNRawHits();
+				nRawHitsR       = (*clusterIter)->getNRawHitsR();
+				nRawHitsPhi     = (*clusterIter)->getNRawHitsPhi();
+				nClusteringType = (*clusterIter)->getClusteringType();
+
+				StFstHit *newHit = new StFstHit(disk, wedge, sensor, apv, charge, chargeErr, maxTb, meanRStrip, meanPhiStrip, nRawHits, nRawHitsR, nRawHitsPhi);
+				newHit->setId(key);
+				newHit->setIdTruth(idTruth);
+
+                                int moduleIdx;
+				if(disk == 1) // Disk 1
+					moduleIdx = wedge;
+				if(disk == 2)// Disk 2
+					moduleIdx = wedge-12;
+				if(disk == 3)// Disk 3
+					moduleIdx = wedge-24;
+				if(disk == 1 || disk == 3)
+				{// Disk 1 & 3
+					phiInner = kFstphiStart[moduleIdx-1]*TMath::Pi()/6.0 + 0.5*kFstzDirct[moduleIdx-1]*kFstStripPitchPhi;
+					phiOuter = kFstphiStop[moduleIdx-1]*TMath::Pi()/6.0  - 0.5*kFstzDirct[moduleIdx-1]*kFstStripPitchPhi;
+				}
+				if(disk == 2)
+				{ // Disk 2
+					phiInner = kFstphiStop[moduleIdx-1]*TMath::Pi()/6.0  - 0.5*kFstzDirct[moduleIdx-1]*kFstStripPitchPhi;
+					phiOuter = kFstphiStart[moduleIdx-1]*TMath::Pi()/6.0 + 0.5*kFstzDirct[moduleIdx-1]*kFstStripPitchPhi;
+				}
+				double local[3];
+				if(meanRStrip < kFstNumRStripsPerWedge/2)
+				{ // inner
+					local[0] = kFstrStart[meanRStrip] + 0.5*kFstStripPitchR;
+					local[1] = phiInner + kFstzFilp[disk-1]*kFstzDirct[moduleIdx-1]*meanPhiStrip*kFstStripPitchPhi;
+				}
+				else
+				{// outer
+					if(sensor == 1){
+						local[0] = kFstrStart[meanRStrip] + 0.5*kFstStripPitchR;
+						local[1] = phiOuter - kFstzFilp[disk-1]*kFstzDirct[moduleIdx-1]*meanPhiStrip*kFstStripPitchPhi - kFstzFilp[disk-1]*kFstzDirct[moduleIdx-1]*0.5*kFstStripGapPhi;
+					}
+					if(sensor == 2){
+						local[0] = kFstrStart[meanRStrip] + 0.5*kFstStripPitchR;
+						local[1] = phiOuter - kFstzFilp[disk-1]*kFstzDirct[moduleIdx-1]*meanPhiStrip*kFstStripPitchPhi + kFstzFilp[disk-1]*kFstzDirct[moduleIdx-1]*0.5*kFstStripGapPhi;
+					}
+				}
+
```

Comment:
These are hardcoded preliminary position of disks.
I think the local[2] should be 0. The transformation to global position should be
be using the DB entries even if it is an ideal entry.

Reply:
@sunxuhit probably needs to comment on this one and follow whatever we decide for other places where DB is needed - either to update now or wait for a future revision.

Reply:
The geometry table is not ready to be uploaded. 
It is better to run the current code first and we will update the table in a later version.